Assets/Scripts/PlayerMovementScript.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+
+[RequireComponent(typeof(CharacterController))]
+
+public class PlayerMovementScript : MonoBehaviour
+{
+    // Movement variables
+    private Vector2 _input;
+    private CharacterController _characterController;
+    private Vector3 _direction;
+    public float _moveSpeed;
+
+    // Gravity variables
+    private float _gravity = -9.81f;
+    public float gravityMultiplyer = 3.0f;
+    private float _velocity;
+    private float _terminalVelocity = -30;
+
+    // Rotation variables
+    [SerializeField] private float smoothTime = 0.05f;
+    private float _currentVelocity;
+
+    void Start()
+    {
+        if (_characterController == null)
+        {
+            _characterController = GetComponent<CharacterController>();
+        }
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
+        _input = new Vector2(Input.GetAxis("Horizontal"), Input.GetAxis("Vertical")).normalized;
+        _direction = new Vector3(_input.x, 0f, _input.y);
```

Comment:
We should probably in the future make sure the direction is based on the camera so pressing forward moves you away from the camera